pkg/envoy/xds/server.go

```diff
@@ -383,19 +386,20 @@ func (s *Server) processRequestStream(ctx context.Context, streamLog *logrus.Ent
 
 			responseLog.Debugf("sending xDS response with %d resources", len(resp.Resources))
 
+			versionStr := strconv.FormatUint(resp.Version, 10)
 			out := &envoy_api_v2.DiscoveryResponse{
-				VersionInfo: strconv.FormatUint(resp.Version, 10),
+				VersionInfo: versionStr,
 				Resources:   resources,
 				Canary:      resp.Canary,
 				TypeUrl:     state.typeURL,
-				Nonce:       nonce,
+				Nonce:       versionStr,
 			}
 			err := stream.Send(out)
 			if err != nil {
 				return err
 			}
 
-			state.nonce = nonce
+			state.nonce = resp.Version
```

Comment:
It seems like `state.nonce` is not used at all anymore. Just remove that field.

Reply:
Removed.